admin_guide/cluster_management.rst

```diff
@@ -354,37 +354,53 @@ Results:
-  │ 2530480378 │  16384 │ 16385 │ select * from foo where id = $1               │ router        │ 42            │     1 │
-  │ 3233520930 │  16384 │ 16385 │ insert into foo select generate_series($1,$2) │ insert-select │ NULL          │     1 │
-  └────────────┴────────┴───────┴───────────────────────────────────────────────┴───────────────┴───────────────┴───────┘
-
-We can see that Citus used the real-time executor to run the count query. This executor fragments the query into constituent queries to run on each node and combines the results on the coordinator node. In the case of the second query (filtering by the distribution column ``id = $1``), Citus determined that it needed the data from just one node. Thus Citus chose the router executor to send the whole query to that node for execution. Lastly, we can see that the ``insert into foo select…`` statement ran with the insert-select executor which provides flexibility to run these kind of queries.  The insert-select executor must sometimes pull results to the coordinator and push them back down to workers.
-
-So far the information in this view doesn't give us anything we couldn't already learn by running the ``EXPLAIN`` command for a given query. However in addition to getting information about individual queries, the ``citus_stat_statements`` view allows us to answer questions such as "what percentage of queries in the cluster are run by which executors?"
+  -[ RECORD 1 ]-+----------------------------------------------
+  queryid       | -6844578505338488014
+  userid        | 10
+  dbid          | 13340
+  query         | SELECT count(*) FROM foo;
+  executor      | adaptive
+  partition_key |
+  calls         | 1
+  -[ RECORD 2 ]-+----------------------------------------------
+  queryid       | 185453597994293667
+  userid        | 10
+  dbid          | 13340
+  query         | insert into foo select generate_series($1,$2)
+  executor      | insert-select
+  partition_key |
+  calls         | 1
+  -[ RECORD 3 ]-+----------------------------------------------
+  queryid       | 1301170733886649828
+  userid        | 10
+  dbid          | 13340
+  query         | SELECT * FROM foo WHERE id = $1
+  executor      | adaptive
+  partition_key | 42
+  calls         | 1
+
+We can see that Citus uses the adaptive executor most commonly to run queries. This executor fragments the query into constituent queries to run on relevant nodes, and combines the results on the coordinator node. In the case of the second query (filtering by the distribution column ``id = $1``), Citus determined that it needed the data from just one node. Lastly, we can see that the ``insert into foo select…`` statement ran with the insert-select executor which provides flexibility to run these kind of queries.  The insert-select executor must sometimes pull results to the coordinator and push them back down to workers.
```

Comment:
> The insert-select executor must sometimes pull results to the coordinator and push them back down to workers.

in fact, the insert-select executor **always** pulls the results to the coordinator. The `INSERT ..SELECT` pushdown goes through the adaptive executor:

```SQL
EXPLAIN INSERT INTO test  SELECT * FROM test;
┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                       QUERY PLAN                                                        │
├─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Custom Scan (Citus Adaptive)  (cost=0.00..0.00 rows=0 width=0)                                                          │
│   Task Count: 32                                                                                                        │
│   Tasks Shown: One of 32                                                                                                │
│   ->  Task                                                                                                              │
│         Node: host=localhost port=9700 dbname=postgres                                                                  │
│         ->  Insert on test_102105 citus_table_alias  (cost=0.00..61.00 rows=13 width=4)                                 │
│               ->  Seq Scan on test_102105 test  (cost=0.00..61.00 rows=13 width=4)                                      │
│                     Filter: ((worker_hash(a) >= '-2147483648'::integer) AND (worker_hash(a) <= '-2013265921'::integer)) │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
(8 rows)
```

The same query, but added `LIMIT 10`, which requires coordinator `INSERT .. SELECT` because the results needs to be `merge`d on the coordinator:
```SQL
 EXPLAIN INSERT INTO test  SELECT * FROM test LIMIT 10;
┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                            QUERY PLAN                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Custom Scan (Citus INSERT ... SELECT via coordinator)  (cost=0.00..0.00 rows=0 width=0)          │
│   ->  Limit  (cost=0.00..0.00 rows=0 width=0)                                                    │
│         ->  Custom Scan (Citus Adaptive)  (cost=0.00..0.00 rows=0 width=0)                       │
│               Task Count: 32                                                                     │
│               Tasks Shown: One of 32                                                             │
│               ->  Task                                                                           │
│                     Node: host=localhost port=9700 dbname=postgres                               │
│                     ->  Limit  (cost=0.00..0.14 rows=10 width=4)                                 │
│                           ->  Seq Scan on test_102105 test  (cost=0.00..35.50 rows=2550 width=4) │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
(9 rows)

```